lib/array.rb

```diff
@@ -0,0 +1,6 @@
+class Array
+  def fadi(element = self.shift, &block)
+    self.each{|n| element = block.call(element,n) }
```

Comment:
Space missing after comma.<br>Space missing to the left of {.<br>Space between { and | missing.